lib/macho/macho_file.rb

```diff
@@ -424,14 +478,24 @@ def low_fileoff
       offset
     end
 
+    # Updates the number of load commands in the raw data.
+    # @param ncmds [Fixnum] the new number of commands
+    # @return [void]
+    # @private
+    def set_ncmds(ncmds)
```

Comment:
`ncmds=` sounds perfect to me. I'm aware of how unidiomatic setter-style methods are in Ruby, but I wasn't sure if it was considered good practice to have an attribute/setter pair like `ncmds` and `ncmds=`. If that's fine, I'm 100% behind it.



> Another thing I'm wondering about is whether set_ncmds and set_sizeofcmds should be merged to a update_header method that always updates both

I think this would be nice (fewer moving parts!), although there are a few use cases where only `sizeofcmds` is changed (dylib/id/rpath changes). I'd like to keep them separated for the time being, but it's definitely worth revisiting as we refactor `MachOFile`.